tests: kernel: Add description for tests in common, interrupt and boot_page_table

spoorthik opened the pull request:
Add description and doxygen links for kernel test cases

Signed-off-by: Spoorthi K <spoorthi.k@intel.com>

nashif commented:
looks good, but we need to add doxygen groups to contain all of those tests.

nashif commented on tests/kernel/arm_irq_vector_table/src/arm_irq_vector_table.c:
```diff
@@ -64,14 +55,23 @@ void isr2(void)
 
 /**
- * @brief test installation of ISRs directly in the vector table
+ * @defgroup kernel_interrupt Interrupt_tests
```
should add a suffix (_tests) to the group to differentiate this from the actual feature documentation, so this becomes kernel_interrupt_tests

nashif replied:
should be 

`@defgroup kernel_interrupt Interrupts`

first argument is the identifier, the second is the title in the documentation

nashif commented on tests/kernel/boot_page_table/src/boot_page_table.c:
```diff
@@ -82,7 +82,22 @@ static void ending_start_addr_range(u32_t start_addr_range)
 }
+/**
+ * @brief Test for boot page table
+ * @defgroup kernel_boot_page_table boot_page_table_tests
```
`@defgroup kernel_boot_page_table_tests Boot Page Table`

We probably want to put this in another category with other tests.

nashif commented on tests/kernel/common/src/main.c:
```diff
@@ -36,7 +36,18 @@ void test_printk(void)
 #endif
 
+/**
+ * @defgroup kernel_common common_tests
```
@defgroup kernel_common_tests Common Tests

nashif commented:
Also see https://github.com/zephyrproject-rtos/zephyr/wiki/Documenting-Code-for-Traceability

spoorthik commented:
@nashif : Addressed review comments. Please have a look.

nashif commented on tests/kernel/arm_runtime_nmi/src/arm_runtime_nmi.c:
```diff
@@ -30,6 +26,14 @@ static void nmi_test_isr(void)
 
+/**
+ * @brief Test the behavior of CONFIG_RUNTIME_NMI at runtime.
+ * @addtogroup kernel_interrupt
```
kernel_interrupt_tests

nashif commented on tests/kernel/common/src/atomic.c:
```diff
@@ -7,6 +7,20 @@
 #include <atomic.h>
 
+/**
+ * @addtogroup kernel_common
```
kernel_common_tests

nashif commented on tests/kernel/common/src/bitfield.c:
```diff
@@ -14,6 +14,19 @@
 #define BITFIELD_SIZE   512
 
+/**
+ * @addtogroup kernel_common
```
kernel_common_tests, and everything below.

spoorthik committed:
tests: kernel: Add description for common and interrrupt

Add test description, RTM links and doxygen links for common,
    interrupt and boot page table test cases.

Signed-off-by: Spoorthi K <spoorthi.k@intel.com>

nashif commented on tests/kernel/common/src/main.c:
```diff
@@ -20,6 +20,14 @@ extern void test_timeout_order(void);
 extern void test_multilib(void);
+
+/**
+ * @defgroup kernel_common_tests common Tests
```
s/common Tests/Common Tests/

nashif commented on tests/kernel/common/src/main.c:
```diff
@@ -20,6 +20,14 @@ extern void test_timeout_order(void);
+/**
+ * @defgroup kernel_common_tests common Tests
+ * @ingroup all_tests
+ * @{
```
when you do this, everything is added to doxygen, also non-tests, either move the helper functions out of the brackets or use `@ingroup kernel_common_tests` individually on each documented test. Here for example delayed_resubmit_work_handler  needs to be moved out or excluded.

spoorthik replied:
@nashif : I think there is no such helper functions to be moved out. All are test functions.

spoorthik commented:
@nashif: Can you please review latest changes?
